openedx/frontend-app-course-authoring
# [FC-0059] feat: Add content sidebar in library home

## Description

This PR adds:
- Library Sidebar with feature of "squash the page"
- Add new content sidebar.
- New button, disabled if the user doesn't have permissions

## Supporting information

- Github issue: https://github.com/openedx/frontend-app-course-authoring/issues/1040
- Related PR: https://github.com/openedx/edx-platform/pull/34934
    - We need the create content in library permission to enable or disable `New` button.
- Internal ticket: [FAL-3754](https://tasks.opencraft.com/browse/FAL-3754)

## Testing instructions

- Enable [library authoring](https://github.com/openedx/openedx-tutor-plugins/tree/main/plugins/tutor-contrib-library-authoring-mfe) in your tutor.
- Use this branch of edx-platform: https://github.com/openedx/edx-platform/pull/34934
- Create and login a new staff user.
- Go to the [library authoring](http://apps.local.edly.io:3001/library-authoring/) and create an new library
- Go to the library home using the id of your new library: http://apps.local.edly.io:2001/course-authoring/library/{library-id}
- Click in `New`
- Create a component. Verify that the component count increases.
- Also you can check the new component on the [library authoring](http://apps.local.edly.io:3001/library-authoring/).
- Enable public read of the library in cms bash:
```
from openedx.core.djangoapps.content_libraries.models import ContentLibrary
lib = ContentLibrary.objects.all()[0]
lib.allow_public_read = True
lib.save()
```
- Logout
- Create and login with a new normal user (not staff).
- Go to the library home using the id of your new library: http://apps.local.edly.io:2001/course-authoring/library/{library-id}
- Verify that the `New` button is disabled


**Settings**

```
EDX_PLATFORM_REPOSITORY: https://github.com/open-craft/edx-platform.git
EDX_PLATFORM_VERSION: taxonomy-sandbox-20240614

GROVE_REDIRECTS:
 - domain: {{ LMS_HOST }}
 - domain: cms.{{ LMS_HOST }}
 - domain: app.{{ LMS_HOST }}
 - domain: meilsearch.{{ LMS_HOST }}

PLUGINS:
- mfe
- grove
- s3
- meilisearch
```

**Tutor requirements**

```
git+https://github.com/overhangio/tutor.git@nightly
git+https://github.com/overhangio/tutor-discovery.git@nightly
git+https://github.com/overhangio/tutor-ecommerce.git@nightly
git+https://github.com/overhangio/tutor-xqueue.git@nightly
git+https://github.com/overhangio/tutor-forum.git@nightly
git+https://gitlab.com/opencraft/dev/tutor-contrib-grove.git@main

git+https://github.com/open-craft/tutor-mfe.git@7d1e1e0dad2c0e4247a4fd8280c257f13ed28432
git+https://github.com/open-craft/tutor-contrib-s3.git@04c7491442fa231e79840995f1f8707eed6353fd
```
